### PR TITLE
[UI] Added a check to handle cases where the input is of type 'object'

### PR DESCRIPTION
**Pull request by agl29**
## What changes were proposed in this pull request?

- Added a check to handle cases where the input is of type 'object'
- Convert the object to a string using `toString()` and `JSON.stringify()` methods

## How was this patch tested?

- Unittest

### CODE REVIEW

File: desktop/core/src/desktop/js/utils/html/deXSS.test.ts [hunk 1]
```diff
@@ -66,4 +66,39 @@ describe('deXSS.ts', () => {
   it('should dump an uppercase javascript url', () => {
     expect(deXSS('<a href="JAVASCRIPT:alert(\'foo\')">Hax</a>')).toEqual('<a>Hax</a>');
   });
+
+  it('should return JSON string for an array', () => {
```
Comment (JohanAhlen) [inline desktop/core/src/desktop/js/utils/html/deXSS.test.ts:70]: Nit: Not really a json string if it doesn't include the [ and ]

File: desktop/core/src/desktop/js/utils/html/deXSS.test.ts [hunk 2]
```diff
@@ -66,4 +66,39 @@ describe('deXSS.ts', () => {
...
+    const nestedObject = {
+      key1: 'value1',
+      key2: {
+        nestedKey: 'nestedValue'
+      }
+    };
+    expect(deXSS(nestedObject)).toEqual('{"key1":"value1","key2":{"nestedKey":"nestedValue"}}');
+  });
+
+  it('should return JSON string for a map containing a struct with arrays', () => {
```
Comment (JohanAhlen) [inline desktop/core/src/desktop/js/utils/html/deXSS.test.ts:94]: Nit: "struct" is just an object in js

**Review by JohanAhlen — APPROVED** [review]
LGTM!

**Review by ananya-agarwal — APPROVED** [review]
LGTM!!

**Review by JohanAhlen — APPROVED** [review]
LGTM!

**Review by Harshg999 — APPROVED** [review]

**Pull request merged**